### DISCUSSION TITLE
Bagaimana cara menampilkan dateModified di Mecha CMS?

### DISCUSSION THREAD

**Discussion by bungfrangki** [mecha-cms/mecha#290] [Q&A] [answered]
Mas, kalau mau manggil dateModified dengan format seperti ini `<?=eat($page->time->format('c'))?>` gimana ya?

**Comment by taufik-nurrohman** [comment, marked as answer]
Tambahkan ini di `.\lot\y\*\index.php`:

``` php
// Buat metode `dateModified`
Page::_('dateModified', function ($mod = null) {
    if (!$this->_exist) {
        return new Time($mod); // Berkas tidak eksis
    }
    $mod = filemtime($this->path);
    return new Time($mod);
});
```

Kemudian di berkas _layout_ bisa dipanggil seperti ini:

``` php
<?= eat($page->dateModified->format('c')); ?>
```

**Reply by bungfrangki** [reply]
saya sudah cobakan mas, tapi error disisi saya.
saya sudah coba juga bikin seperti ini:
`<?= eat($page->time->format('c'),filemtime(eat($url->current(false, false)))); ?>`

memang tidak error, tapi tanggal tetap sama dengan _date time published_

**Reply by taufik-nurrohman** [reply]
Jangan pakai `$url->current()` mas itu kan rute publik, bukan rute berkas.

**Reply by taufik-nurrohman** [reply]
> tapi tanggal tetap sama dengan _date time published_

Ya kalau laman belum pernah diperbarui ya nilainya bakalan sama dengan waktu publikasi.

Untuk `$page->time` juga baiknya mengambil data dari berkas `time.data` supaya nilainya konstan. Tanpa itu nilainya akan mengambil dari `filectime($page->path)` yang kadang nilainya tidak konsisten.

**Comment by taufik-nurrohman** [comment]
> saya sudah cobakan mas, tapi error disisi saya.

Sudah ketemu masalahnya mas, karena kurang nilai _default_ pada argumen metode. Di sini Saya mengeset nilai _default_ menjadi `null`:

``` php
Page::_('dateModified', function ($mod = null) {});
```

**Reply by bungfrangki** [reply]
berhasil mas!
terima kasih banget (y)
